### PR TITLE
Optimize "see more" action on navigation filter

### PR DESCRIPTION
**Pull request by thomas-kl1**

### CODE REVIEW

**Comment by romainruaud** [comment]
How could the call to ``$this->categoryRepository->create()`` work if this is not a factory anymore but the repository directly ?

**Comment by thomas-kl1** [comment]
Whoops missed it in my copy paste!

**Comment by romainruaud** [comment]
Could you please squash your commits and reduce their content to the modification related to the CategoryRepository ? You modified the cosmetic of the whole file but it's not necessary and merging it as is would flag too much lines as being concerned by the CategoryRepository refactoring.

Regards

File: src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php [hunk 1]
```diff
@@ -185,7 +178,7 @@ private function getItems()
      */
     private function isSearch()
     {
-        return (bool) ($this->getRequest()->getParam('q') !== null);
```
Comment (thomas-kl1) [inline src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php:188]: Useless cast to bool

File: src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php [hunk 2]
```diff
@@ -165,11 +156,13 @@ private function getItems()
...
         $layer = $this->layerResolver->get();
         $filterList = $this->getFilterList();
+        $filterName = (string)$this->getFilterName();
+        $baseUrl = $this->_redirect->getRedirectUrl();
 
         foreach ($filterList->getFilters($layer) as $filter) {
-            if ($filter->getRequestVar() == $this->getFilterName()) {
+            if ($filter->getRequestVar() === $filterName) {
                 foreach ($filter->getItems() as $item) {
-                    $item->setBaseUrl($this->_redirect->getRedirectUrl());
```
Comment (thomas-kl1) [inline src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php:172]: getRedirectUrl is built at every iteration

File: src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php [hunk 3]
```diff
@@ -78,10 +78,7 @@ public function execute()
     {
         $this->initLayer();
 
-        $items  = $this->getItems();
-        $result = $this->jsonResultFactory->create()->setData($items);
```
Comment (thomas-kl1) [inline src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php:82]: unnecessary variables

File: src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php [hunk 4]
```diff
@@ -53,14 +53,14 @@ class Ajax extends \Magento\Framework\App\Action\Action
...
      * @param \Magento\Catalog\Model\Layer\Resolver                   $layerResolver      Layer resolver.
-     * @param \Magento\Catalog\Api\CategoryRepositoryInterfaceFactory $categoryRepository Category factory.
+     * @param \Magento\Catalog\Api\CategoryRepositoryInterface        $categoryRepository Category Repository.
      * @param \Magento\Catalog\Model\Layer\FilterList[]               $filterListPool     Filter list pool.
      */
     public function __construct(
         \Magento\Framework\App\Action\Context $context,
         \Magento\Framework\Controller\Result\JsonFactory $jsonResultFactory,
         \Magento\Catalog\Model\Layer\Resolver $layerResolver,
-        \Magento\Catalog\Api\CategoryRepositoryInterfaceFactory $categoryRepository,
```
Comment (thomas-kl1) [inline src/module-elasticsuite-catalog/Controller/Navigation/Filter/Ajax.php:63]: inject shared instance of the category repository so we can benefice from entities loaded within the registry from the layer

**Comment by thomas-kl1** [comment]
@romainruaud I just rephrased the PR and squashed the commit. I've added comments regarding the changes.

**Comment by romainruaud** [comment]
Thanks, can you fix the PHPCs violations so that we can start reviewing and testing this ?

Best regards

**Pull request merged**